### DISCUSSION TITLE
2.7 对象抽象

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#112] [Announcements]
# 2.7 对象抽象

SICP Python 中文翻译

https://composingprograms.netlify.app/2/7

**Comment by wqj2004** [comment]
第二章翻译也太不用心了吧...第一章光靠翻译还能读的下去，第二章光看翻译就完全读不下去而必须对照英文原文了....而且还有许多错别字...

**Reply by CharlieCRX** [reply]
其实有个大概印象就好，这节内容挺恶心的，感觉需要反复阅读才行

**Reply by zhou427** [reply]
确实好多错译和缺字，只能边看边PR了hh

**Reply by CharlieCRX** [reply]
加油bro！

**Reply by sleepy-shawn** [reply]
第一次听repr, str那里没听懂，现在准备再听一遍

**Reply by ghost** [reply]
可以直接把网站扔给ai让他润色文档😂

**Comment by kv-chiu** [comment]
搜泛型搜到这一节，没看懂，失意地去看了typescript的泛型讲解，感觉好了一点

**Comment by null1213** [comment]
接口的概念还是不太懂，有没有大神解释一下

**Reply by von-Neumann101** [reply]
接口就是标准：比如列表和字典，他们是可迭代的，接口就是让程序知道，他们是可迭代的，你实现了接口（一个可以迭代的对象的标准），具体说实现了接口定义的方法，就是跟解释器证明了你是可迭代的。

**Comment by N0rth-W1nd** [comment]
2.7.3节这段话中第三句：“Complex类存储了...”原文是“ComplexRI class stores...”，应为“ComplexRI类存储了...”：

Python 有一种简单的计算属性的特性，可以通过零参数函数实时的计算属性。@property 修饰符允许函数在没有调用表达式语法（表达式后跟随圆括号）的情况下被调用。Complex 类存储了 real 和 imag 属性并在需要时计算 magnitude 和 angle 属性。

**Comment by SwordKillCowardice** [comment]
这一节后两部分看的似懂非懂的 和第一章高阶函数那里一毛一样的

**Comment by Alexababab** [comment]
看得似懂非懂，可能还是要干lab hk来理解到位了
